utils/terraform_client.py

```diff
@@ -23,81 +26,144 @@ def __init__(self, body, integration, integration_version):
-                    logging.info(['replace', name, line_split[0],
-                                  line_split[1].split(' ', 1)[0]])
-                else:
-                    logging.error(['replace', name, line_split[0],
-                                   line_split[1].split(' ', 1)[0]])
-                    logging.error('\'replace\' action is not enabled. ' +
-                                  'Please run the integration manually ' +
-                                  'with the \'--enable-deletion\' flag.')
-                deletions_detected = True
+        with self._log_lock:
+            for line in stdout:
+                line = line.strip()
+                if line.startswith('+ aws'):
+                    line_split = line.replace('+ ', '').split('.')
+                    logging.info(['create', name, line_split[0],
+                                  line_split[1]])
+                if line.startswith('- aws'):
+                    line_split = line.replace('- ', '').split('.')
+                    if enable_deletion:
+                        logging.info(['destroy', name,
+                                      line_split[0], line_split[1]])
+                    else:
+                        logging.error(['destroy', name,
+                                       line_split[0], line_split[1]])
+                        logging.error('\'destroy\' action is not enabled. ' +
+                                      'Please run the integration manually ' +
+                                      'with the \'--enable-deletion\' flag.')
+                    deletions_detected = True
+                if line.startswith('~ aws'):
+                    line_split = line.replace('~ ', '').split('.')
+                    logging.info(['update', name, line_split[0],
+                                  line_split[1]])
+                if line.startswith('-/+ aws'):
+                    line_split = line.replace('-/+ ', '').split('.')
+                    if enable_deletion:
+                        logging.info(['replace', name, line_split[0],
+                                      line_split[1].split(' ', 1)[0]])
+                    else:
+                        logging.error(['replace', name, line_split[0],
+                                       line_split[1].split(' ', 1)[0]])
+                        logging.error('\'replace\' action is not enabled. ' +
+                                      'Please run the integration manually ' +
+                                      'with the \'--enable-deletion\' flag.')
+                    deletions_detected = True
         return deletions_detected
 
+    # terraform apply
     def apply(self):
         errors = False
-        for name, tf in self.tfs.items():
-            return_code, stdout, stderr = tf.apply(auto_approve=True)
-            error = self.check_output(name, return_code, stdout, stderr)
+
+        pool = ThreadPool(1)  # TODO: replace 1 with self.thread_pool_size
```

Comment:
TODO

Reply:
to remain TODO until we gain more experience :)